test/test_scan.cpp

```diff
@@ -175,4 +180,160 @@ boost::compute::exclusive_scan(
     CHECK_RANGE_EQUAL(int, 4, output, (0, 1, 3, 6));
 }
 
+BOOST_AUTO_TEST_CASE(inclusive_scan_int_multiplies)
+{
+//! [inclusive_scan_int_multiplies]
+// setup input
+int data[] = { 1, 2, 1, 2, 3 };
+boost::compute::vector<int> input(data, data + 5, queue);
+
+// setup output
+boost::compute::vector<int> output(5, context);
+
+// inclusive scan with multiplication
+boost::compute::inclusive_scan(
+    input.begin(), input.end(), output.begin(),
+    boost::compute::multiplies<int>()
+);
+
+// output = [1, 2, 2, 4, 12]
+//! [inclusive_scan_int_multiplies]
+
+    BOOST_CHECK_EQUAL(input.size(), size_t(5));
+    BOOST_CHECK_EQUAL(output.size(), size_t(5));
+
+    CHECK_RANGE_EQUAL(int, 5, output, (1, 2, 2, 4, 12));
+
+    // in-place inclusive scan
+    CHECK_RANGE_EQUAL(int, 5, input, (1, 2, 1, 2, 3));
+    boost::compute::inclusive_scan(input.begin(), input.end(), input.begin(),
+                                   boost::compute::multiplies<int>());
```

Comment:
Same here, make sure you pass the `queue` argument so that the calls to `inclusive_scan` and the code to check the values are all on the same `command_queue`.